content/articles/how-to-build-a-random-password-generator-app-with-html-css-javascript/index.md

```diff
@@ -0,0 +1,298 @@
+
+```JavaScript
+// Generating Character Codes For The Application
+const UPPERCASE_CODES = arrayFromLowToHigh(65, 90);
+const LOWERCASE_CODES = arrayFromLowToHigh(97, 122);
+const NUMBER_CODES = arrayFromLowToHigh(48, 57);
+const SYMBOL_CODES = arrayFromLowToHigh(33, 47)
+  .concat(arrayFromLowToHigh(58, 64))
+  .concat(arrayFromLowToHigh(91, 96))
+  .concat(arrayFromLowToHigh(123, 126));
+```
+
+The next thing you want to do is build the `copy` button and copy to clipboard functionality. We will listen to the click event on the “copybtnDOM” element. Meanwhile, inside that function, you will create a `textarea` element using the `createElement` method in javascript.
+
+```JavaScript
+// Copy Password Button
+
+copybtnDOM.addEventListener("click", () => {
+  const textarea = document.createElement("textarea");
+  const passwordToCopy = resultDOM.innerText;
+  // A Case when Password is Empty
+  if (!passwordToCopy) return;
+  // Copy Functionality
+  textarea.value = passwordToCopy;
+  document.body.appendChild(textarea);
+  textarea.select();
+  document.execCommand("copy");
+  textarea.remove();
+  alert("Password Copied to Clipboard");
+});
+```
+
+In the above code for `const textarea = document.createElement(‘textarea’)`, you just created a variable that holds the textarea DOM element. You can get the value for the textarea using the `resultDOM.innerText`.
+
+Then, the `if (!passwordToCopy) return;` means if the “passwordToCopy” variable is empty, we should do nothing. The `textarea.value = passwordToCopy` syntax is used to set the “textarea” value with the value you want to copy.
+
+Next is the `document.body.appendchild(textarea);` all you want to do here is append the text area value to the body of your document. The append child method adds a node to the end of the list of the children of a specified parent node. 
+
+You need to select the elements you want to copy so you will use the `textarea.select();` method then to copy the elements, you will use the `document.execCommand(‘copy’);` method. This method executes the specified command for the selected part of an editable section which is the textarea. The `copy` command inside the function will copy the values of the editable section, after which you need to remove the “textarea” by using the `textarea.remove();`. This will give the user a notification that the password is successfully copied. 
+
+After that, you will code a “simple alert” for it to show it, the `alert(‘password copied to clipboard’)` function. With all these written down, the application can not yet generate random passwords because you have not finished implementing the functionality that creates random passwords.
```

Comment:
Why are you discussing the copy functionality before completing the generate password?
Please move this after the section where you explain the generate password function.